src/Providers/Mistral/Mistral.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EchoLabs\Prism\Providers\Mistral;
+
+use EchoLabs\Prism\Contracts\Provider;
+use EchoLabs\Prism\Enums\FinishReason;
+use EchoLabs\Prism\Exceptions\PrismException;
+use EchoLabs\Prism\Providers\ProviderResponse;
+use EchoLabs\Prism\Requests\TextRequest;
+use EchoLabs\Prism\ValueObjects\ToolCall;
+use EchoLabs\Prism\ValueObjects\Usage;
+use Throwable;
+
+class Mistral implements Provider
+{
+    protected Client $client;
+
+    protected string $model;
+
+    public function __construct(
+        public readonly string $url,
+        public readonly ?string $apiKey,
```

Comment:
We should make the API key required

Reply:
@sixlive  Pushed the changes. [9802f42](https://github.com/echolabsdev/prism/pull/13/commits/9802f423c1b91da29b93098baab58e8e505dd8b5)